gtk: add resize overlay

jcollie opened the pull request:
This adds a transient overlay that shows the size of the surface while you are resizing the window or the surfaces.

[Screencast from 2024-08-09 21-20-20.webm](https://github.com/user-attachments/assets/d75ee853-c44b-49ef-8d14-cdb517a29d8a)

jcollie committed:
gtk: add resize overlay

This adds a transient overlay that shows the size of the surface
while you are resizing the window or the surfaces.

mitchellh reviewed (commented):
Really awesome. I didn't dig into the GTK APIs but did a high level review with some requests but this is exciting to see!

jcollie committed:
resize-overlay: change default and delay → duration

Change the default to `after-first` and change `resize-overlay-delay` →
`resize-overlay-duration`.

jcollie committed:
resize overlay: move all resize overlay code and data into a struct

Keepin´ the code clean by gathering all of the resize overlay
related data and code into a struct.

jcollie committed:
move resize overlay code to a new file to keep the file size down

mitchellh committed:
apprt/gtk: small style tweaks for resize overlay
